easybuild/easyblocks/h/healpix.py

```diff
@@ -62,11 +72,32 @@ def configure_step(self):
         if not cfitsio:
             raise EasyBuildError("Failed to determine root for CFITSIO, module not loaded?")
 
+        # target:
+        #   1: basic_gcc
+        #   2: generic_gcc
+        #   3: linux_icc
+        #   4: optimized_gcc
+        #   5: osx
+        #   6: osx_icc
+
         self.comp_fam = self.toolchain.comp_family()
         if self.comp_fam == toolchain.INTELCOMP:  # @UndefinedVariable
-            cxx_config = '4'  # linux_icc
+            cxx_config = '3'  # linux_icc
+            self.target_string = 'linux_icc'
         elif self.comp_fam == toolchain.GCC:  # @UndefinedVariable
-            cxx_config = '2'  # generic_gcc
+            if self.cfg['gcc_target'] == 'basic_gcc':
+                cxx_config = '1'
+                self.target_string = 'basic_gcc'
+            elif self.cfg['gcc_target'] == 'generic_gcc':
+                cxx_config = '2'
+                self.target_string = 'generic_gcc'
+            elif self.cfg['gcc_target'] == 'optimized_gcc':
+                cxx_config = '4'
+                self.target_string = 'optimized_gcc'
+            else:
+                # by default let's go with generic_gcc:
```

Comment:
This should be an error imho, to protect against undesired behavior in case a silly typo is made...